src/app/shared/filter/podcasts.component.ts

```diff
@@ -0,0 +1,45 @@
+import { Component, OnInit, OnDestroy } from '@angular/core';
+import { Store } from '@ngrx/store';
+import { Subscription } from 'rxjs/Subscription';
+import { selectPodcastFilter, selectPodcasts } from '../../ngrx/reducers';
+import { PodcastModel } from '../../ngrx/model';
+import { CastleFilterAction } from '../../ngrx/actions';
+
+@Component({
+  selector: 'metrics-podcasts',
+  template: `
+    <prx-select *ngIf="allPodcastOptions.length > 1"
+                single="true" searchable="true"
+                [options]="allPodcastOptions" [selected]="selectedPodcast" (onSelect)="onPodcastChange($event)">
+    </prx-select>
+    <span *ngIf="allPodcastOptions.length === 1 && selectedPodcast">{{ selectedPodcast.title }}</span>
+  `
+})
+export class PodcastsComponent implements OnInit, OnDestroy {
+  podcastFilterSubscription: Subscription;
+  podcastsSubscription: Subscription;
+  selectedPodcast: PodcastModel;
+  allPodcastOptions = [];
+
+  constructor(public store: Store<any>) {}
+
+  ngOnInit() {
+    this.podcastFilterSubscription = this.store.select(selectPodcastFilter).subscribe((filteredPodcast: PodcastModel) => {
```

Comment:
oooh, I'm a little late to the party but this createSelector business is awesome!

Reply:
immutability rules still apply, but can be handy when you want a smaller/deeper slice of state